src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = sourceString.split(';');
+
+  const finalResult = {};
+
+  result.forEach((element) => {
+    const [key, value] = element.split(':');
+
+    if (key && value) {
+      finalResult[key.trim()] = value.trim();
+    }
+  });
```

Comment:
[CODE KNOWLEDGE] - You don't need to make a copy before each operation. The `split` method does not mutate the original string, and the `forEach` method is an exception to the rule against modifying an existing array/object. You can directly use `sourceString.split(';')` in the `forEach` loop.